baseplate/experiments/__init__.py

```diff
@@ -77,6 +78,16 @@ def _get_config(self, name):
             logger.warning("Could not load experiment config: %s", str(exc))
             return
 
+    def _get_experiment(self, name):
+        if name not in self._experiment_cache:
```

Comment:
Maybe that's a dangerous assumption to make though?  FWIW I've been thinking about the caching a bit more and I'm not really sure that we _need_ it.  It only shaves 20 ~ 30 microseconds off the `variant` time.  That's not nothing, especially in cases like the listing endpoints in r2 where we do over 1k feature checks, but even in those cases it's generally not a huge difference.  We have to do 3333 ~ 5000 feature checks before that adds up to 100 milliseconds.